ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -0,0 +1,185 @@
+    cancel(event);
+    trackEvent({
+      category: EVENT.CATEGORIES.TRANSACTIONS,
+      event: 'Cancel',
+      properties: {
+        action: 'SIWE Request',
+        type,
+        version,
+      },
+    });
+  };
+
+  const generateSIWEWarning = () => {
+    let showError = false;
+    let errorMessage = '';
+    let confirmErrorMessage = '';
+
+    if (!isSIWEDomainValid) {
+      showError = true;
+      errorMessage += `${t('SIWEDomainInvalid', [messageData.domain])} `;
+      confirmErrorMessage += `${t('SIWEDomainWarningBody', [
+        messageData.domain,
+      ])} `;
+    }
+
+    if (!isMatchingAddress) {
+      showError = true;
+      errorMessage += `${t('SIWEAddressInvalid', [
+        messageData.address,
+        fromAccount.address,
+      ])} `;
+      confirmErrorMessage += `${t('SIWEAddressInvalid', [
+        messageData.address,
+        fromAccount.address,
+      ])} `;
+    }
+
+    return {
+      showError,
+      errorMessage: errorMessage.trim(),
+      confirmErrorMessage: confirmErrorMessage.trim(),
+    };
+  };
+
+  const {
+    showError,
+    errorMessage,
+    confirmErrorMessage,
+  } = generateSIWEWarning();
+
+  console.log(showError, errorMessage, confirmErrorMessage);
+
+  return (
+    <div className="signature-request-siwe">
+      <Header
+        fromAccount={fromAccount}
+        domain={messageData.domain}
+        isSIWEDomainValid={isSIWEDomainValid}
+        subjectMetadata={subjectMetadata}
+      />
+      <Message data={formatMessageParams(messageData, t)} />
+      {showError && (
+        <div className="signature-request-siwe__domain-mismatch-warning">
+          <ErrorMessage errorMessage={errorMessage} />
+        </div>
+      )}
+      <PageContainerFooter
+        footerClassName="signature-request-siwe__page-container-footer"
+        onCancel={onCancel}
+        onSubmit={showError ? () => setIsShowingDomainWarning(true) : onSign}
+        cancelText={t('cancel')}
+        submitText={t('signin')}
+      />
+      {isShowingDomainWarning && (
+        <Popover
+          onClose={() => setIsShowingDomainWarning(false)}
+          title={t('SIWEWarningTitle')}
+          subtitle={t('SIWEWarningSubtitle')}
+          className="signature-request-siwe__popover"
+          footerClassName="signature-request-siwe__popover__footer"
+          footer={
+            <PageContainerFooter
+              footerClassName="signature-request-siwe__popover__footer__warning-footer"
+              onCancel={() => setIsShowingDomainWarning(false)}
+              cancelText={t('cancel')}
+              cancelButtonType="default"
+              onSubmit={onSign}
+              submitText={t('confirm')}
+              submitButtonType="danger-primary"
+              disabled={!agreeToDomainWarning}
+            />
+          }
+        >
+          <div className="signature-request-siwe__popover__checkbox-wrapper">
+            <Checkbox
+              id="domainWarning_checkbox"
```

Comment:
```suggestion
              id="signature-request-siwe_domain-checkbox"
```
To avoid name collisions, I am suggesting that we include the name of the fileName in the id name. Reminder that will need to update `htmlFor` below with the name as well

Reply:
Addressed.